lib/connectors/memory.js

```diff
@@ -148,11 +148,13 @@ Memory.prototype.saveToFile = function (result, callback) {
   if (file) {
     if(!self.writeQueue) {
       // Create a queue for writes
+      var cache = self.cache;
+      var ids = self.ids;
       self.writeQueue = async.queue(function (task, cb) {
         // Flush out the models/ids
         var data = JSON.stringify({
-          ids: self.ids,
-          models: self.cache
+          ids: ids,
+          models: cache
```

Comment:
@bajtos Yes too many things are happening in this PR; probbaly I will move it to a standalone PR